application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep

```diff
@@ -0,0 +1,412 @@
+param publicIpRGName string = resourceGroup().name
+
+@description('Select Image Deployment for debugging only')
+@allowed([
+  'development'
+  'production'
+])
+param environment string = 'production'
+
+@description('Tags by resource.')
+param outTagsByResource object = {}
+
+@description('The base URI where artifacts required by this template are located including a trailing \'/\'')
+param _artifactsLocation string = deployment().properties.templateLink.id
+
+@description('The sasToken required to access _artifactsLocation.')
+@secure()
+param _artifactsLocationSasToken string = ''
+
+@description('Enable or disable Unreal Pixel Streaming port.')
+param unrealPixelStreamingEnabled bool = false
+
+@description('Enable Managed Identity')
+param enableManagedIdentity bool = false
+
+@description('Enable Azure Active Directory Login')
+param enableAAD bool = false
+
+var environmentMapping = {
+  ue_4_27: 'unreal_4_27'
+  ue_5_0ea: 'unreal_5_0ea'
+  unity_2020_3_19f1: 'unity_2020_3_19f1'
+}
+
+var environments = {
+  development: {
+    vmImage: {
+      publisher: 'microsoft-agci-gaming'
+      offer: 'agci-gamedev-image'
+      sku: 'gamedev-${gameEngine}-${osType}'
+      version: 'latest'
+    }
+    vmPlan: {
+      publisher: 'microsoft-agci-gaming'
+      product: 'agci-gamedev-image'
+      name: 'gamedev-${gameEngine}-${osType}'
+    }
+  }
+  production: {
+    vmImage: {
+      publisher: 'microsoftcorporation1602274591143'
+      offer: 'game-dev-vm'
+      sku: '${osType}_${environmentMapping[gameEngine]}'
+      version: 'latest'
+    }
+    vmPlan: {
+      publisher: 'microsoftcorporation1602274591143'
+      product: 'game-dev-vm'
+      name: '${osType}_${environmentMapping[gameEngine]}'
+    }
+  }
+}
+
+var vmImage = environments[environment].vmImage
+var vmPlan = environments[environment].vmPlan
+
+var ipconfName = '${vmName}-ipconf'
+var nicName = '${vmName}-nic'
+var nsgName = '${vmName}-nsg'
+
+var storageType = (bool(length(split(vmSize, '_')) > 2) ? 'Premium_LRS' : 'Standard_LRS')
+
+var tags = {
+  'solution': 'Game Development Virtual Machine'
+  'engine': gameEngine
+  'ostype': osType
+  'remotesoftware': remoteAccessTechnology
+}
+
+var cmdGDKInstall = '(Get-Content \'C:\\ProgramData\\Microsoft\\Windows\\Start Menu\\Programs\\Startup\\gdkinstall.cmd\').replace(\'[VERSION]\', \'${gdkVersion}\') | Set-Content \'C:\\ProgramData\\Microsoft\\Windows\\Start Menu\\Programs\\Startup\\gdkinstall.cmd\''
+var userData = 'team_id=${parsec_teamId}:key=${parsec_teamKey}:name=${parsec_host}:user_email=${parsec_userEmail}:is_guest_access=${parsec_isGuestAccess}:ibLicenseKey=${ibLicenseKey}'
+var Script2Run = 'TeradiciRegCAS.ps1'
+var CSEParams = ' -pcoip_registration_code ${teradiciRegKey}'
+var cmdTeradiciRegistration = './${Script2Run}${CSEParams}'
+
+var vnetId = {
+  'new'     : resourceId('Microsoft.Network/virtualNetworks', vnetName)
+  'existing': resourceId(vnetRGName, 'Microsoft.Network/virtualNetworks', vnetName)
```

Comment:
this syntax will work for both new and existing, since you default the RG name...  so you can flatten this var

Reply:
> this syntax will work for both new and existing, since you default the RG name... so you can flatten this var

`var publicIpId = publicIpNewOrExisting == 'none' ? '' : resourceId(publicIpRGName, 'Microsoft.Network/publicIPAddresses', publicIpName)`